File: python-examples/17-order-processing/worker.py
"""Worker for order processing example."""

import asyncio
import os

from dotenv import load_dotenv
from polos import PolosClient, Worker

from tools import charge_stripe, send_confirmation_email
from agents import order_agent
from workflows import order_processing_workflow

load_dotenv()


async def main():
    """Run the worker."""
    project_id = os.getenv("POLOS_PROJECT_ID")
    if not project_id:
        raise ValueError("POLOS_PROJECT_ID environment variable is required")

    client = PolosClient(
        project_id=project_id,
        api_url=os.getenv("POLOS_API_URL", "http://localhost:8080"),
    )

    worker = Worker(
        client=client,
        deployment_id="order-processing",
        workflows=[order_processing_workflow],
        agents=[order_agent],
        tools=[charge_stripe, send_confirmation_email],
    )

    print("Starting order processing worker...")
    print(f"  Project ID: {project_id}")
    print("  Press Ctrl+C to stop\n")

    try:
        await worker.run()
    except KeyboardInterrupt:
        print("\nShutting down...")
        await worker.shutdown()


if __name__ == "__main__":
    asyncio.run(main())

File: typescript-examples/17-order-processing/worker.ts
/**
 * Polos Worker for the Order Processing example.
 *
 * Run with:
 *   npx tsx worker.ts
 *
 * Environment variables:
 *   POLOS_PROJECT_ID - Your project ID (required)
 *   POLOS_API_URL - Orchestrator URL (default: http://localhost:8080)
 *   POLOS_API_KEY - API key for authentication (optional for local development)
 *   POLOS_DEPLOYMENT_ID - Deployment ID (default: order-processing)
 *   OPENAI_API_KEY - OpenAI API key (required for the agent)
 */

import 'dotenv/config';
import { Worker } from '@polos/sdk';
import { chargeStripe, sendConfirmationEmail } from './tools.js';
import { orderAgent } from './agents.js';
import { orderProcessingWorkflow } from './workflows.js';

async function main() {
  const projectId = process.env['POLOS_PROJECT_ID'];
  if (!projectId) {
    throw new Error(
      'POLOS_PROJECT_ID environment variable is required. ' +
        'Set it to your project ID (e.g., export POLOS_PROJECT_ID=my-project). ' +
        'You can get this from the output printed by `polos-server start` or from the UI page at ' +
        "http://localhost:5173/projects/settings (the ID will be below the project name 'default')",
    );
  }

  const apiUrl = process.env['POLOS_API_URL'] ?? 'http://localhost:8080';
  const apiKey = process.env['POLOS_API_KEY'] ?? '';
  const deploymentId = process.env['POLOS_DEPLOYMENT_ID'] ?? 'order-processing';

  const worker = new Worker({
    apiUrl,
    apiKey,
    projectId,
    deploymentId,
    workflows: [orderProcessingWorkflow, orderAgent, chargeStripe, sendConfirmationEmail],
  });

  console.log('Starting Order Processing worker...');
  console.log(`  Project ID: ${projectId}`);
  console.log(`  Workflows: [${[orderProcessingWorkflow, orderAgent, chargeStripe, sendConfirmationEmail].map((w) => w.id).join(', ')}]`);
  console.log('  Press Ctrl+C to stop\n');

  await worker.run();
}

main().catch(console.error);
